frontend/views/components/ContributionItem.vue

```diff
@@ -0,0 +1,161 @@
+      },
+      default: 'RECEIVING'
+    },
+    type: {
+      type: String,
+      validator: function (value) {
+        // The value must match one of these strings
+        return ['NON_MONETARY', 'MONETARY'].indexOf(value) !== -1
+      },
+      default: 'NON_MONETARY'
+    }
+  },
+  data () {
+    return {
+      isVisible: false
+    }
+  },
+  computed: {
+    contributionText () {
+      if (this.hasWhoElse) {
+        const html = {
+          service: `<span class="has-text-bold">${this.what}</span>`,
+          button_: '<button class="is-unstyled is-link-inherit link">',
+          numMembers: this.otherContributor,
+          _button: '</button>'
+        }
+
+        if (this.action === 'RECEIVING' && this.type === 'MONETARY') {
+          return L('{service} from {button_}{numMembers} members{_button}', html)
+        } else {
+          return L('A total of {service} to {button_}{numMembers} members{_button}', html)
+        }
+      } else {
+        if (this.action === 'RECEIVING') {
+          const html = {
+            service: `<span class="has-text-bold">${this.what}</span>`,
+            who: this.firstWho
+          }
+          return L('{service} by {who}', html)
+        } else {
+          if (this.type === 'MONETARY') {
+            const html = {
+              amount: `<span class="has-text-bold">${this.what}</span>`,
+              who: this.firstWho
+            }
+            return L('{amount} to {who}', html)
+          } else {
+            return `<span class="has-text-bold">${this.what}</span>`
+          }
+        }
+      }
+    },
+
+    listOfName () {
+      const html = {
+        amount: `<span class="has-text-bold">${this.what}</span>`,
+        listName: this.who.map((name, index) => {
+          return `<p class="has-text-1 c-contribution-list-item">${name}</p>`
+        }).join('')
+      }
+      if (this.action === 'RECEIVING') {
+        return L('{amount} from {listName}', html)
+      } else {
+        return L('A total of {amount} to {listName}', html)
+      }
+    },
+
+    firstWho () {
+      const who = this.who
+      if (!Array.isArray(who)) return who
+      return who.length === 2 ? this.L('{who0} and {who1}', { who0: who[0], who1: who[1] }) : who[0]
+    },
+
+    otherContributor () {
+      return this.who.length // (-1 contributor + 1 array length)
+    },
+
+    hasWhoElse () {
+      return Array.isArray(this.who) && this.who.length > 2
+    },
+    iconClass () {
+      const style = {
+        'NON_MONETARY': {
+          icon: 'heart',
+          color: 'warning'
+        },
+        'MONETARY': {
+          icon: 'dollar-sign',
+          color: 'success'
+        }
+      }
+      return `icon-${style[this.type].icon} c-icon-round has-background-${style[this.type].color} has-text-${style[this.type].color}`
+    }
+  }
+}
+</script>
+
+<style lang="scss">
+@import "@assets/style/_variables.scss";
+
+.c-contribution-list-item {
+  &:nth-child(2) {
```

Comment:
@pieer, you can use `::v-deep` to make a style "global". In this case:

```scss
::v-deep .c-contribution-list-item { ... }
```
It should work even when placed in "scoped" <style>, so you can delete this style.